src/lib/ffi/ffi_pkey_algs.cpp

```diff
@@ -990,32 +1066,29 @@ int botan_privkey_load_kyber(botan_privkey_t* key, const uint8_t privkey[], size
 
 int botan_pubkey_load_kyber(botan_pubkey_t* key, const uint8_t pubkey[], size_t key_len) {
 #if defined(BOTAN_HAS_KYBER)
+   if(key == nullptr) {
+      return BOTAN_FFI_ERROR_NULL_POINTER;
+   }
    *key = nullptr;
-   switch(key_len) {
-      case 800:
-         return ffi_guard_thunk(__func__, [=]() -> int {
-            const std::vector<uint8_t> pubkey_vec(pubkey, pubkey + 800);
-            auto kyber512 = std::make_unique<Botan::Kyber_PublicKey>(pubkey_vec, Botan::KyberMode::Kyber512_R3);
-            *key = new botan_pubkey_struct(std::move(kyber512));
-            return BOTAN_FFI_SUCCESS;
-         });
-      case 1184:
-         return ffi_guard_thunk(__func__, [=]() -> int {
-            const std::vector<uint8_t> pubkey_vec(pubkey, pubkey + 1184);
-            auto kyber768 = std::make_unique<Botan::Kyber_PublicKey>(pubkey_vec, Botan::KyberMode::Kyber768_R3);
-            *key = new botan_pubkey_struct(std::move(kyber768));
-            return BOTAN_FFI_SUCCESS;
-         });
-      case 1568:
-         return ffi_guard_thunk(__func__, [=]() -> int {
-            const std::vector<uint8_t> pubkey_vec(pubkey, pubkey + 1568);
-            auto kyber1024 = std::make_unique<Botan::Kyber_PublicKey>(pubkey_vec, Botan::KyberMode::Kyber1024_R3);
-            *key = new botan_pubkey_struct(std::move(kyber1024));
-            return BOTAN_FFI_SUCCESS;
-         });
-      default:
-         BOTAN_UNUSED(key, pubkey, key_len);
-         return BOTAN_FFI_ERROR_BAD_PARAMETER;
+
+   const auto mode = [](size_t len) -> std::optional<Botan::KyberMode> {
+      if(len == 800) {
+         return Botan::KyberMode::Kyber512_R3;
+      } else if(len == 1184) {
+         return Botan::KyberMode::Kyber768_R3;
+      } else if(len == 1568) {
+         return Botan::KyberMode::Kyber1024_R3;
+      } else {
+         return {};
+      }
+   }(key_len);
+
+   if(mode.has_value()) {
+      auto kyber = std::make_unique<Botan::Kyber_PublicKey>(std::span{pubkey, key_len}, *mode);
+      *key = new botan_pubkey_struct(std::move(kyber));
+      return BOTAN_FFI_SUCCESS;
+   } else {
+      return BOTAN_FFI_ERROR_BAD_PARAMETER;
```

Comment:
As above. Purely opinionated nitpick: Early return on error followed by the actual "happy path" at the very end of the function.